### PR TITLE
frontend: Add progress bars to Start Installation page

### LINKED ISSUE #562: Add progress bars to progress screen
For the apply step, this will be timed to the average deploy time and stop at 80% if we run over. This is because Terraform gives us no status info.

For downloading and starting pods, we will use the embedded Kubernetes (https://github.com/coreos/tectonic-installer/issues/561) client to measure progress.

### PR DESCRIPTION
**Pull request by kyoto**
Adds progress bars for both the AWS Terraform apply step and for the Starting Tectonic step for both AWS and bare metal.

The Terraform progress is estimated by parsing the log output. This PR uses a very simple approach with the intention of improving the progress bar accuracy in a future PR.

The Tectonic progress is based on the sub-steps, except that the progress bar will not move backwards of a sub-step flips off again.

Makes some CSS changes to improve horizontal alignment of the page elements.

Adds back the install step "cancel" state, introduced by #1274, which seems to have been
accidentally removed and changes its text color to match the pending state.

Fixes #562

screenshot-1
screenshot-1
screenshot-3
screenshot-4

### CODE REVIEW

**Comment by tlwu2013** [comment]
Hi @kyoto , 

Looking good, thanks!
Just want to paste our discussions on Slack here for tracking:

**[Tony]**
one quick feedback is that 
_for this “Start Installation” screen, “Starting Tectonic” step should be activated after “Terraform apply success”_.
that way, there should be only one progress bar on the screen

not sure if making ” _‘Starting Tectonic’ step should be activated after ‘Terraform apply success’_ ” should be separated out into a different PR or done in this PR.

But if it’s going to be a separate PR, then I feel it’ll be better if we merge that logic first, then merge this progress bar PR.

Thanks Andy!

**[Andy]**
ok, I’ll add that change to this PR
thanks!

**Comment by kyoto** [comment]
Thanks @tlwu2013. I made that change and updated the screenshots.

**Comment by kyoto** [comment]
@tlwu2013 Updated again. I also applied the "inactive" style to the "Resolving..." step, which I think better matches the logic of what the installation is doing.

**Comment by kyoto** [comment]
ok to test

File: installer/frontend/components/tf-poweron.jsx [hunk 1]
```diff
@@ -13,6 +13,20 @@ import { CLUSTER_NAME, PLATFORM_TYPE, getTectonicDomain } from '../cluster-confi
...
+const ProgressBar = ({progress}) => <div className="wiz-launch-progress__bar">
+  <div className="wiz-launch-progress__bar-inner" style={{width: `${progress * 100}%`}}></div>
+</div>;
+
+const estimateTerraformProgress = terraform => {
+  const {output = ''} = terraform;
+  const done = (output.match(/: Creation complete/g) || []).length;
+
+  // Approximate number of AWS resources we expect Terraform to create
+  const total = 145;
```
Comment (ggreer) [inline installer/frontend/components/tf-poweron.jsx]: Doesn't the number of resources to create depend on the size of the cluster? Would larger clusters claim to be completed far sooner than they actually are?

File: installer/frontend/components/ui.jsx [hunk 2]
```diff
@@ -597,23 +597,26 @@ export const PrivateKeyArea = (props) => {
...
+export const WaitingLi = ({pending, done, error, cancel, children, substep}) => {
   const progressClasses = classNames({
     'wiz-launch-progress__step': !substep,
     'wiz-launch-progress__substep': substep,
+    'wiz-pending-fg': pending,
     'wiz-error-fg': error,
     'wiz-success-fg': done && !error,
-    'wiz-running-fg': !done && !error,
+    'wiz-cancel-fg': !done && !error && cancel,
+    'wiz-running-fg': !done && !error && !cancel && !pending,
```
Comment (ggreer) [inline installer/frontend/components/ui.jsx]: Since pending & cancel are styled the same and the cancel parameter isn't used anywhere besides this PR, would it make sense to subsume them both into a parameter & style called inactive? eg:

```jsx
 <WaitingLi inactive={terraformRunning || tfError} done={allDone} error={anyFailed} key="tectonicReady">
```

Comment (kyoto) [inline installer/frontend/components/ui.jsx]: Actually, the style is different. They have the same text color, but different icons.

Comment (ggreer) [inline installer/frontend/components/ui.jsx]: Oh. Nevermind then.

**Comment by tlwu2013** [comment]
Hey @kyoto , 

Progress bar looks great in terms of styling, thanks!

**Comment by kyoto** [comment]
Rebased on master.

File: installer/frontend/components/tf-poweron.jsx (hunk 1 above)
Comment (kyoto) [inline installer/frontend/components/tf-poweron.jsx]: Yeah, I tried getting a more accurate number my looking at the number of instances, but that turned out to not be so helpful. It needs more experimentation, but I'd like to do that in a separate PR.

**Comment by kyoto** [comment]
ok to test

**Comment by kyoto** [comment]
ok to test

**Comment by kyoto** [comment]
ok to test
